### ISSUE TITLE
How to Force a Re-Render?

### ISSUE THREAD

**Issue by Xigtim** [handsontable/vue-handsontable-official#30]
Hi Everyone,

I'm trying to make the grid re-render, however I cant figure out how to do it.

According to the official documentation I need to reference the Handsontable object. But I cant access it from the Handsontable Vue Component.

['Understand Binding as a Reference':](https://docs.handsontable.com/0.35.0/tutorial-data-binding.html)
```javascript
  hot1 = new Handsontable(container1, settings1);
  data1[0][1] = 'Ford';
  hot1.render();
```

 Any suggestions?

Thanks in advance!

**Comment by superpig** [comment]
same problem +1

**Comment by Xigtim** [comment]
Help please, I really need to know how to do this.
Thanks!

**Comment by qiumobai** [comment]
same problem +1

**Comment by tonypolik** [comment]
I think it's not documented, but the component has a `table` property which is a reference to the Handsontable object. See [here](https://github.com/handsontable/vue-handsontable-official/blob/master/src/helpers.js#L52).

Assuming you have access to the HotTable component via `$refs`, you can do something like:

```javascript
this.$refs.myHotTableRef.table.render()
```

**Comment by xeoshow** [comment]
Also met the same problem. 
How can we create a "new Handsontable" via js code? Currently from the sampleApp.vue, it is just done in template part as below:
------
    <div id="hot-preview">
      <HotTable :root="root" :settings="hotSettings"></HotTable>
    </div>
------

Just would like to know the counterpart of the js code ...

Thanks !

**Comment by titan83** [comment]
Hello, guys.
@tonypolik show us right direction, but so lamer like me)) don't see all picture. But after "smoking" vue docs https://vuejs.org/v2/guide/components.html#Child-Component-Refs I did next hack for rerendering:
`<HotTable :root="root" :settings="hotSettings" ref="myTable"></HotTable>`
Here ref prop is very important.
We will use it in js code:
this.hotSettings.colHeader[0] = 'We changed it!'
this.$refs.myTable.table.render()
That's all.
Hope this  small post helps to somebody.

**Comment by httpete** [comment]
Struggling, I must not understand something core.  I have the ref to my hotTable. but it won't render the array of objects.


`  mounted() {
   
    let vm = this;
      this.$feathers
        .service("teams")
        .find()
        .then(d => {
           vm.$refs.testHot.table.data = d.data;
           vm.$refs.testHot.table.render();
           debugger;
        });


  },`

**Comment by tylerkoldenjtp** [comment]
@tonypolik thanks for the tip!  Your solution only worked for me if I passed a string to the render function (I didn't try passing a function).  According to the [documentation ](https://docs.handsontable.com/pro/1.18.1/Options.html#renderer) I used 'numeric' because all of my fields were number fields.

`this.$refs.myTable.table.render('numeric')`

**Comment by mrpiotr-dev** [comment]
Hi @Zigtim,
Here is a simple demo, how you can do it: https://codepen.io/swistach/pen/NMzmPa

@tonypolik is right. It's not documented enough good, and, as in his example, `this.$refs.myTable.table` is a reference to `Handsontable.Core`, so you can use each method from `Core`:https://docs.handsontable.com/pro/2.0.0/Core.html

**Comment by avanlaningham** [comment]
I am running hansontable 4.0 and I could not get this to work.  I found this article 
https://handsontable.com/docs/7.0.3/frameworks-wrapper-for-vue-vuex-example.html
which inspried me to change the above examples from this.$refs.myTable.table
to this.$refs.myTable.hotInstance  which I could then run .render() on.

**Comment by AMBudnik** [comment]
Hey @avanlaningham as this topic is a year old can I ask you to share a demo with your recent progress? If so, please send it at support@handsontable.com

**Comment by tanthammar** [comment]
@avanlaningham 
Confirmed.
This works:
```js
methods: {
    redrawTable() {
        this.$refs.myTable.hotInstance.render();
    },
```

**Comment by AMBudnik** [comment]
Thank for you confirming @tanthammar

**Comment by EthanYan6** [comment]
> @avanlaningham
> Confirmed.
> This works:
> 
> ```js
> methods: {
>     redrawTable() {
>         this.$refs.myTable.hotInstance.render();
>     },
> ```

I can find the function,but it don't work for me

**Comment by AMBudnik** [comment]
Hi @EthanYan6 here's a `render()` called within `methods` https://jsfiddle.net/AMBudnik/7qox8c3b/
I've added a log to the `afterRender()` hook so you can track if it is actually called.
If that won't work for you please send me your code sample to support@handsontable.com

**Comment by EthanYan6** [comment]
> track if it is actually called.

First of all, thank you for your reply. This hook is indeed executed, but when the data is reloaded, it will have an impact, that is, the last data remains. I temporarily solved it with the following code and achieved the effect I wanted:

```vue
            this.$refs.hotTable.hotInstance.loadData([]);
            this.$refs.hotTable.hotInstance.render();
```

My idea is: first empty the data, then use render(), and finally fill the table with new data. Although I spared a lot of time, I finally achieved the result I wanted
